tests/scans/dependencies/README.md

```diff
@@ -1,22 +1,24 @@
 # Dependencies Scanner
 
 ## Description
-It's a tool used to scan for vulnerabilities in a requirements.txt file.\
-It can generate reports via console output or json file. Can be run with `pytest` and manage to handle remote files under github repositories. Requirements file can be specified with `repo`, `branch`, `requirements-path` parameters giving flexibility on file location.
-Output file in which the report will be generated can be specified with `report-path` parameter.
+
+The Dependencies Scanner is a tool for scanning vulnerabilities in a *requirements.txt* file from different GitHub repositories. 
+It uses `pytest` to run and can generate reports via console output or JSON files.
+
+The requirements file to scan can be specified with the `repo`, `reference`, and `requirements-path` parameters. 
+Moreover, the path of the report file generated can be chosen with the `report-path` parameter.
```

Comment:
At least, one of the problems reported in the issue was still not addressed. According to [the issue](https://github.com/wazuh/wazuh-qa/issues/4274#issue-1787927980): 

> No Support for comments and spaces in a requirements file:
> Ex, this will cause an error:
> 
> ```
> # AWS
> boto3==1.17.85
> 
> # Azure
> azure-storage-blob==2.1.0
> ```

This is still happening so the test should be fixed or the README updated to mention the expected format ([error req file](https://github.com/wazuh/wazuh/blob/test-deps-scan-requirements/framework/requirements.txt)):
<details>
<summary>Error in requirement.txt with comments</summary>

```
$ python3 -m pytest -vv -x --disable-warnings dependencies/ --repo wazuh --reference test-deps-scan-requirements --requirements-path framework/requirements.txt --report-path ~/Escritorio/test_file.json
============================================================================================ test session starts =============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.0.0 -- /home/selu/venv/wazuh-qa/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.10.12', 'Platform': 'Linux-5.15.0-83-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.1.2', 'pluggy': '1.0.0'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '2.0.4'}}
rootdir: /home/selu/Git/wazuh-qa/tests, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-2.0.4
collected 1 item                                                                                                                                                                                             

dependencies/test_dependencies.py::test_python_dependencies_vuln_scan FAILED                                                                                                                           [100%]

================================================================================================== FAILURES ==================================================================================================
_____________________________________________________________________________________ test_python_dependencies_vuln_scan _____________________________________________________________________________________

pytestconfig = <_pytest.config.Config object at 0x7f2a31421750>

    def test_python_dependencies_vuln_scan(pytestconfig):
        """Check that the specified dependencies do not have any known vulnerabilities.
    
        Args:
            pytestconfig (fixture): Fixture that returns the :class:`_pytest.config.Config` object.
        """
        branch = pytestconfig.getoption('--reference')
        repo = pytestconfig.getoption('--repo')
        requirements_path = pytestconfig.getoption('--requirements-path')
        report_path = pytestconfig.getoption('--report-path')
        requirements_url = f"https://raw.githubusercontent.com/wazuh/{repo}/{branch}/{requirements_path}"
        urlretrieve(requirements_url, REQUIREMENTS_TEMP_FILE.name)
>       result = report_for_pytest(REQUIREMENTS_TEMP_FILE.name,
                                   os.path.join(os.path.dirname(os.path.abspath(__file__)), "known_flaws_deps.json"))

dependencies/test_dependencies.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/selu/venv/wazuh-qa/lib/python3.10/site-packages/wazuh_testing/tools/scans/dependencies.py:119: in report_for_pytest
    prepare_input(False, requirements_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pip_mode = False, input_file_path = '/tmp/tmpyuy8vigj'

    def prepare_input(pip_mode, input_file_path):
        """Create temp input file with all packages listed and prepared to be scanned later on.
    
        Args:
            pip_mode (bool): enable/disable pip freeze to retrieve package information.
            input_file_path (str): path to the input file (used if pip_mode is disabled).
        """
        python_process = subprocess.run([python_bin, '--version'], stdout=subprocess.PIPE, universal_newlines=True)
        pkg = python_process.stdout.strip().split()
        package_list.append(package_tuple(pkg[0], pkg[1]))
        if pip_mode:
            pip_mode_process = subprocess.run([python_bin, '-m', 'pip', 'freeze'], stdout=subprocess.PIPE,
                                              universal_newlines=True)
            for package_line in pip_mode_process.stdout.strip().split('\n'):
                pkg = package_line.strip().split('==')
                package_list.append(package_tuple(pkg[0], pkg[1]))
        else:
            with open(input_file_path, mode='r') as input_file:
                lines = input_file.readlines()
                for line in lines:
                    line = re.sub('[<>~]', '=', line)
                    if ',' in line:
                        package_version = max(re.findall(r'\d+\.+\d*\.*\d', line))
                        package_name = re.findall('([a-z]+)', line)[0]
                        line = f'{package_name}=={package_version}\n'
                    if ';' in line:
                        line = line.split(';')[0] + '\n'
                    pkg = line.strip().split('==')
>                   package_list.append(package_tuple(pkg[0], pkg[1]))
E                   IndexError: list index out of range

/home/selu/venv/wazuh-qa/lib/python3.10/site-packages/wazuh_testing/tools/scans/dependencies.py:74: IndexError
========================================================================================== short test summary info ===========================================================================================
FAILED dependencies/test_dependencies.py::test_python_dependencies_vuln_scan - IndexError: list index out of range
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================= 1 failed in 0.59s ==============================================================================================
```

</details>